### ISSUE TITLE
Filter challenges by year / start date

### ISSUE THREAD

**Issue by tschaffter** [Sage-Bionetworks/sage-monorepo#1039]
## Tasks

- [x] Add the properties `Challenge.startDate` and `Challenge.endDate`
- [x] Filter challenges by year

## Current UI

<img width="336" alt="image" src="https://user-images.githubusercontent.com/3056480/209419050-948040a2-c05e-4f1e-b6e6-f653b0031695.png">

## Notes

- We could add an "information icon" with a tooltip that specifies that "Challenge Year" refers to the challenge start year.
- We don't have the exact date of most past DREAM Challenges. We could set by default the date to the January 1st 00:00 and include this time has the lower bracket for the date range filter.

**Comment by tschaffter** [comment]
## Spring Boot does not support query parameter with `deepObject`?

Swagger UI generates correctly the request URL when replacing the default:

```java
      @Parameter(
              name = "startDateRange",
              description = "Return challenges that start during the date range specified.")
``` 

by

```java
      @Parameter(
              name = "startDateRange",
              description = "Return challenges that start during the date range specified.",
              style = ParameterStyle.DEEPOBJECT,
              explode = Explode.TRUE)
```

However, the value of the properties `start` and `end` are not set correctly and instead are set to null.

**Comment by tschaffter** [comment]
Addressed in #1137
